xml_struct/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! This crate provides a mechanism for serializing Rust data structures as
+//! well-formed XML with a minimum of boilerplate.
+//!
+//! Consumers can provide manual implementations of the [`XmlSerialize`] and
+//! [`XmlSerializeAttr`] traits if desired, but the primary intent of this crate
+//! is to provide automated derivation of these traits in order to facilitate
+//! serialization of complex XML structures.
+//!
+//! # Limitations
+//!
+//! At present, derived implementations of these traits are designed to handle
+//! the specific case of Microsoft Exchange Web Services. As such, all XML
+//! elements and attributes are named in PascalCase and certain behaviors are
+//! not supported (such as serializing enum variants without enclosing XML
+//! elements derived from the variant name).
+//!
+//! There is also currently no provision for deserialization from XML, as the
+//! support offered by `quick_xml`'s serde implementation has been found to be
+//! sufficient for the time being.
```

Comment:
I think that limitations can be both temporary and permanent, though I think eventually it would be preferable to cover deserialization with this crate as well.